scaleway/config.go

```diff
@@ -23,16 +25,44 @@ type Config struct {
 	Region       string
 }
 
-// Client contains scaleway api clients
-type Client struct {
-	scaleway *sdk.API
+// Meta contains SDK clients used by resources.
+//
+// This meta value is passed into all resources.
+type Meta struct {
+	// Deprecated: The deprecated Scaleway SDK (will be removed in `v2.0.0`).
+	deprecatedClient *sdk.API
 }
 
-// client is a bridge between sdk.HTTPClient interface and retryablehttp.Client
+// Meta creates a meta instance from a client configuration.
+func (c *Config) Meta() (*Meta, error) {
+	meta := &Meta{}
+
+	deprecatedClient, err := c.GetDeprecatedClient()
```

Comment:
what do you think about moving the code into a package and importing it as `deprecated.Meta()`? This way you don't need to rename everything and can call the deprecated methods in your new provider structs?

(recurs)

Reply:
It's a good idea but:
- Terraform recommends keeping all codebase in the `main` package
- This deprecated code will be replaced in a month at maximum

So we will keep a single package.